schemas/post-artifact-request.yml

```diff
@@ -3,6 +3,49 @@ title:              "Post Artifact Request"
 description: "Request a authorization to put and artifact or posting of a URL as an artifact. Note that the `storageType` property is referenced in the response as well."
 type:               object
 oneOf:
+  - title:          "Blob Artifact Request"
+    description: |
+      Request for to upload a generic blob artifact.
+    type:           object
+    properties:
+      storageType:
+        description: |
+          Artifact storage type, in this case `'blob'`
+        type:       string
+        enum:
+          - blob
+      contentType:
+        description: |
+          Artifact mime-type, when uploading artifact to the signed
+          `PUT` URL returned from this request this must given with the
+           `ContentType` header. Please, provide correct mime-type,
+           this make tooling a lot easier, specifically,
+           always using `application/json` for JSON artifacts.
+        type:       string
+        maxLength:  255
+      sha256:       {type: string, pattern: '^[0-9a-f]{64}$'}
+      md5:          {type: string, pattern: '^[0-9a-f]{32}$'}
+      parts:
+        type:       array
+        items:
+          type:     object
+          properties:
+            sha256:       {type: string, pattern: '^[0-9a-f]{64}$'}
+            md5:          {type: string, pattern: '^[0-9a-f]{32}$'}
+            size:         {type: integer, minimum: 0}            
+          additionalProperties: false
+          required:
+            - sha256
+            - md5
+            - size
+        minItems: 1
+    additionalProperties: false
+    required:
+      - storageType
+      - contentType
+      - sha256
+      - md5
+      - parts
```

Comment:
As I've mentioned before, there's a difference between a single part upload (`PUT` for object) and a multipart upload that has only one part.  I would rather signify that difference by the presence of the parts list.  We'd have to have an extra check that the single part's sha256 matches the file's sha256.  That's not possible in JSON Schema, at least not easily as best as I can tell.  Further, it doesn't match the intent.  There might be some practical reason for using multipart uploads for single part uploads in the future.  I'd rather be explicit about this, especially as figuring out the basics of the upload is a client responsibility.

How about we compromise and have the schema say that presence of the parts list implies multipart upload, and that if specified we ensure that there are two or more items in the list.